pkg/analysis/jsontags/analyzer.go

```diff
@@ -75,15 +76,13 @@ func (a *analyzer) run(pass *analysis.Pass) (any, error) {
 }
 
 func (a *analyzer) checkField(pass *analysis.Pass, field *ast.Field, tagInfo extractjsontags.FieldTagInfo) {
-	var prefix string
+	prefix := "embedded field %s"
 	if len(field.Names) > 0 && field.Names[0] != nil {
-		prefix = fmt.Sprintf("field %s", field.Names[0].Name)
-	} else if ident, ok := field.Type.(*ast.Ident); ok {
-		prefix = fmt.Sprintf("embedded field %s", ident.Name)
+		prefix = "field %s"
 	}
 
 	if tagInfo.Missing {
-		pass.Reportf(field.Pos(), "%s is missing json tag", prefix)
+		pass.Reportf(field.Pos(), "%s is missing json tag", fmt.Sprintf(prefix, utils.FieldName(field)))
```

Comment:
Probably better to call the `FieldName` once/twice when we are setting up the prefix, than three times down here

Reply:
fixed.